src/lib.rs

```diff
@@ -287,15 +290,63 @@ impl error::Error for Error {
             Error::InvalidLength => "invalid length",
             Error::InvalidChar(_) => "invalid character",
             Error::InvalidData(_) => "invalid data point",
+            Error::InvalidPadding => "invalid padding",
             Error::MixedCase => "mixed-case strings not allowed",
         }
     }
 }
 
+type ConvertResult = Result<Vec<u8>, Error>;
+
+/// Convert between bit sizes
+///
+/// # Panics
+/// Function will panic if attempting to convert `from` or `to` a bit size that
+/// is larger than 8 bits.
+///
+/// # Examples
+///
+/// ```rust
+/// use bech32::convert_bits;
+/// let base5 = convert_bits(vec![0xff], 8, 5, true);
+/// assert_eq!(base5.unwrap(), vec![0x1f, 0x1c]);
+/// ```
+pub fn convert_bits(data: Vec<u8>, from: u32, to: u32, pad: bool) -> ConvertResult {
+    if from > 8 || to > 8 {
```

Comment:
What happens if from or to is 0? From looking at the code the function could run indefinitely. Maybe it should panic instead.

Reply:
Yeah, panicking would be better.

Reply:
`panic` on 0 would need a couple additional tests as well

Reply:
opened #11 to fix this, but it would be nice if #10 could be merged first since rebasing is easier for me this way.